### DISCUSSION TITLE
8MB binary files lag UI and how to Set() file name from golang as framework

### DISCUSSION THREAD

**Discussion by navaneeth-dev** [pocketbase/pocketbase#2411] [Q&A] [answered]
So I am storing large binary files and the UI just lags any way to fix?

Second, how to set file name of binary from golang with Set:

Example which is not working rn:
```go
multiPartData, err := createMultipartFormData("../storage/xxx-binary.exe")
record.Set("binary", multiPartData)
```

In UI this doesn't show file name so I cannot download

**Comment by ganigeorgiev** [comment]
I'm not sure I follow. Are you trying to store raw blob as a record field? If Yes, then this is currently not supported. Technically you can store it as `text` field but I'm not sure what the visualization would be.

Could you elaborate a little more on your use case and what you are trying to do?

**Reply by navaneeth-dev** [reply]
I am trying to store a generated exe like github actions after build command and I delete it after the user downloads the file.

So there are 2 issues right now:
1. UI lags due to the binary file being large (is there any option to not display the contents?)
2. How do I store the binary file after `OnRecordAfterCreateRequest` is called. I know we have to use form data to set the file name but it isn't working with the above code.
3. Is it safe to do `go function`
Eg:
![image](https://user-images.githubusercontent.com/32010211/235950853-5a86481b-7570-4c74-87e1-9fc27899822f.png)

![image](https://user-images.githubusercontent.com/32010211/235949525-4bdc202f-086a-42ee-951c-6a6aa0c10d44.png)
![Untitled-2023-05-03-2008](https://user-images.githubusercontent.com/32010211/235950509-2f40c7b0-ce7d-4e0d-8683-9d6306574ea1.png)

Thanks in advance

**Reply by navaneeth-dev** [reply]
EDIT: OK I understood why UI is lagging its because the raw data is being placed in the **file name of admin dashboard**. So yeah how to fix the file name issue?

![image](https://user-images.githubusercontent.com/32010211/235952790-1adaaa97-88f6-47e5-bd02-022fe153617e.png)

**Reply by ganigeorgiev** [reply, marked as answer]
But why do you need to store the file as `text` field in the first place?
Is adding it as regular `file` field not an option?

Here is an example using the `forms.RecordUpsert` to perform also the field validations (_size and mimetime_):

```go
form := forms.NewRecordUpsert(app, record)

// you can also use filesystem.NewFileFromBytes(bytescontent)
file1, _ := filesystem.NewFileFromPath("path/to/file")
file2, _ := filesystem.NewFileFromPath("path/to/file")

form.AddFiles("yourFileField", file1, file2)

err := form.Submit()
```

---

> Is it safe to do go function 

I'm not sure that I understand the provided code sample. Note that if the `createBuild()` goroutine panics there will be no recover for it. If you don't care about the result of `createBuild()`, you can use the helper `routine.FireAndForget(yourFunction)`.

**Reply by navaneeth-dev** [reply]
Thanks a lot! I am using **file only not text**. And now its working.

What is the difference between `routine.FireAndForget(yourFunction)` and `go yourFunction`.

Sry I am new

**Reply by ganigeorgiev** [reply]
> What is the difference between routine.FireAndForget(yourFunction) and go yourFunction.

[`routine.FireAndForget`](https://github.com/pocketbase/pocketbase/blob/master/tools/routine/routine.go) has a built-in panic-recover handling.
